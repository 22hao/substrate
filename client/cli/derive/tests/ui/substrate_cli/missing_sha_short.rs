use sc_cli_derive::substrate_cli;

struct MyCli {}

#[substrate_cli(support_url = "http://example.org", copyright_start_year = 2020)]
impl sc_cli::SubstrateCli for MyCli {
	fn load_spec(&self, id: &str) -> Result<Box<dyn sc_service::ChainSpec>, String> {
		unimplemented!()
	}
}

fn main() {}
