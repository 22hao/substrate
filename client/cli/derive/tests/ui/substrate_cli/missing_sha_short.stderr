error: missing env variable VERGEN_SHA_SHORT. You might want to add a build.rs script that runs substrate_build_script_utils::generate_cargo_keys()
 --> $DIR/missing_sha_short.rs:5:1
  |
5 | #[substrate_cli(support_url = "http://example.org", copyright_start_year = 2020)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
